.buildkite/pipeline.yml

```diff
@@ -1,8 +1,12 @@
+env:
+  RUST_BACKTRACE: short
+  RUSTFLAGS: -D warnings -W unreachable-pub -W rust-2021-compatibility
```

Comment:
`-W unreachable-pub` will probably break every crate of ours. I do think it's a generally good thing to have, but I am not sure if it's good enough to actually enable for our use-case. But this should be a great way to test if this actually works.

Reply:
If we want to poach from rust-analyzer, than `CARGO_INCREMENTAL=0` *might* be interesting avenue to explore to speed up CI: https://matklad.github.io/2021/09/04/fast-rust-builds.html#ci-caching. Needs substantial investigation though